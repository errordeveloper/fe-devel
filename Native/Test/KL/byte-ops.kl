/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

operator entry()
{
  // Unary ops
  report(+Byte(42));
  report(-Byte(42));

  // Binary ops
  report(Byte(1) + Byte(2));
  report(Byte(3) - Byte(4));
  report(Byte(6) * Byte(2));
  report(Byte(6) / Byte(2));
  report(Byte(11) % Byte(5));
  report(Byte(11) << Byte(3));
  report(Byte(11) >> Byte(2));

  // Assignment ops
  var Byte x = 4; report(x);
  x = Byte(5); report(x);
  x += Byte(2); report(x);
  x -= Byte(3); report(x);
  x *= Byte(12); report(x);
  x /= Byte(2); report(x);
  x %= Byte(5); report(x);
  x <<= Byte(1); report(x);
  x >>= Byte(1); report(x);
  
  // Bit ops
  report(Byte(7) & Byte(11));
  report(Byte(7) | Byte(11));
  report(Byte(7) ^ Byte(11));
  
  // Log ops
  report(Byte(0) && Byte(11));
  report(Byte(7) || Byte(0));
}
