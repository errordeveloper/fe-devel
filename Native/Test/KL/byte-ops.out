42
214
3
255
12
3
1
88
2
4
5
7
4
48
24
4
8
4
3
15
12
0
7
